File: security_assistant/remediation/code_examples/sql_injection_fix.js
// VULNERABLE (Node.js mysql):
// const query = "SELECT * FROM users WHERE id = " + userId;
// connection.query(query, (error, results) => { ... });

// SECURE (Node.js mysql):
// Use ? placeholders
const query = "SELECT * FROM users WHERE id = ?";
connection.query(query, [userId], (error, results) => { ... });

// VULNERABLE (Postgres pg):
// client.query("SELECT * FROM users WHERE id = " + userId);

// SECURE (Postgres pg):
// Use $1, $2 placeholders
client.query("SELECT * FROM users WHERE id = $1", [userId]);

File: security_assistant/remediation/code_examples/sql_injection_fix.java
// VULNERABLE:
// String query = "SELECT * FROM users WHERE username = '" + username + "'";
// Statement stmt = connection.createStatement();
// ResultSet rs = stmt.executeQuery(query);

// SECURE (Prepared Statement):
String query = "SELECT * FROM users WHERE username = ?";
PreparedStatement pstmt = connection.prepareStatement(query);
pstmt.setString(1, username);
ResultSet rs = pstmt.executeQuery();
